compiler/rustc_data_structures/src/sync.rs

```diff
@@ -378,50 +378,41 @@ impl<K: Eq + Hash, V: Eq, S: BuildHasher> HashMapExt<K, V> for HashMap<K, V, S>
 }
 
 #[derive(Debug)]
-pub struct Lock<T>(InnerLock<T>);
+pub struct Lock<T>(Mutex<T>);
+
+unsafe impl<T: std::marker::Send> std::marker::Send for Lock<T> {}
```

Comment:
This definitely needs a comment justifying this.

Reply:
Thanks for pointing out!